clients/drcachesim/scheduler/scheduler.h

```diff
@@ -830,6 +836,12 @@ template <typename RecordType, typename ReaderType> class scheduler_tmpl_t {
     scheduler_options_t options_;
     std::vector<input_info_t> inputs_;
     std::vector<output_info_t> outputs_;
+    // We use a central lock for global scheduling.  We assume the synchronization
+    // cost is outweighed by the simulator's overhead.  This protects concurrent
+    // access to the inputs_, outputs_, and ready_ fields.
+    std::mutex sched_lock_;
+    // Input indices ready to be scheduled.
+    std::queue<int> ready_;
```

Comment:
We should typedef a type for input ordinal. It'll make APIs and declarations more readable.

Reply:
Created input_ordinal_t and output_ordinal_t.  This changes the signatures of some previously exported routines but the interfaces is in flux in any case.